### ISSUE TITLE
DesktopAbstractTable.EditableColumnTableCellEditor.getCellEditorValue unnecessarily captures focus

### ISSUE THREAD

**Issue by dkz** [cuba-platform/cuba#1095]
### Environment

- Platform version: 6.8.10
- Client type: Desktop 

### Description of the bug

tl;dr: Desktop Abstract Table requests focus on `.getCellEditorValue` which not only counter intuitive (chaning focus state in getter method), but also breaks one certain use case.
This case is somewhat complex to reproduce in a demo project so I will do my best to provide a comprehensive description of it below.

I use extended version of desktop table component, called TaxiDesktopTable which overrided an `.editingStopped` method of swing extensions table:
```
public class TaxiDesktopTable<E extends Entity> extends DesktopTable<E> {
    public TaxiDesktopTable() {
        //TODO remove after CUBA's updating
        impl = new TaxiJXTableExt() {
            @Override
            public void editingStopped(ChangeEvent e) {
                TableCellEditor editor = getCellEditor();
                if (editor != null) {
                    Object value = editor.getCellEditorValue(); // this one causes nuisance
                    ...
            }
        };
}
```
And also I have an simple editor (important: editor is open in a modal window mode) with a table which goes like this where `availableToService` is a Boolean property:
```
<table editable="true" width="100%" height="100%">
  <columns>
    <column id="service.name" caption="Service"/>
    <column id="availableToService" caption="Enabled" editable="true"/>
  </columns>
  <rows datasource="availablePaymentTypeDs"/>
</table>
```
But the thing is that I want to keep checkbox value intact if a user actions break a certain condition (Dont want to have a service with no payment types available), so I double-check the condition and show a warning dialog when needed:
```
availablePaymentTypeDs.addItemPropertyChangeListener(e -> {
  ServiceAvailablePaymentType item = e.getItem();
  boolean value = BooleanUtils.isTrue(e.getValue());
  if (myConiditionIsKindaNotSatisfied(value, item)) {
    item.setAvailableToService(true);
    showMessageDialog(...);
  }
});
```
And finally clicking on a checkbox raises a warning dialog, but focus jumps to the editor window and therefore it shows blocked editor window with modal message warning underneath the editor window.

This happens because of that `.edditingStopped` method I mentioned before: it calls for `getCellEditorValue` and it efficently brings the focus to the blocked editor window. I it expected for message dialog window to appear on top of the windowed editor.

### Technical note
When I comment focus request out of `DesktopAbstractTable` everything works perfectly fine and as expected:
```
@Override
public Object getCellEditorValue() {
  flush(DesktopComponentsHelper.getComposition(cellComponent));
  // impl.requestFocus(); ← this one
  if (cellComponent instanceof HasValue) {
    return ((Field) cellComponent).getValue();
  }
  return null;
}
```
So I would probably suggest either to remove that that line or request focus only if a component is inside an active window.

**Comment by jreznot** [comment]
@dkz Please check if it works with 6.8-SNAPSHOT and close issue if everything is fine
